src/main/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStore.kt

```diff
@@ -148,24 +151,33 @@ class ParticipantProjectSpeciesStore(
     val existing = fetchOneById(participantProjectSpeciesId)
     val updated = updateFunc(existing)
 
-    dslContext.transaction { _ ->
-      val rowsUpdated =
-          with(PARTICIPANT_PROJECT_SPECIES) {
-            dslContext
-                .update(PARTICIPANT_PROJECT_SPECIES)
-                .set(FEEDBACK, updated.feedback)
-                .set(MODIFIED_BY, currentUser().userId)
-                .set(MODIFIED_TIME, clock.instant())
-                .set(RATIONALE, updated.rationale)
-                .set(SUBMISSION_STATUS_ID, updated.submissionStatus)
-                .where(ID.eq(participantProjectSpeciesId))
-                .execute()
-          }
-
-      if (rowsUpdated < 1) {
-        throw ParticipantProjectSpeciesNotFoundException(participantProjectSpeciesId)
-      }
-    }
+    val modifiedTime = clock.instant()
+
+    val participantProjectSpecies =
+        dslContext
+            .selectFrom(PARTICIPANT_PROJECT_SPECIES)
+            .where(PARTICIPANT_PROJECT_SPECIES.ID.eq(participantProjectSpeciesId))
+            .fetchOne()
+            ?: throw ParticipantProjectSpeciesNotFoundException(participantProjectSpeciesId)
+
+    val oldParticipantProjectSpecies =
+        ExistingParticipantProjectSpeciesModel.of(participantProjectSpecies)
+
+    participantProjectSpecies.feedback = updated.feedback
+    participantProjectSpecies.modifiedBy = currentUser().userId
+    participantProjectSpecies.modifiedTime = modifiedTime
+    participantProjectSpecies.rationale = updated.rationale
+    participantProjectSpecies.submissionStatusId = updated.submissionStatus
```

Comment:
Happy to see someone else using `Record` instead of `Row` classes for this! Might be a little prettier to wrap this in `with(participantProjectSpecies)` so the assignments become `feedback = updated.feedback` and so on, though you'd need to rename the local `modifiedTime` variable to avoid name collision.

Reply:
Pairs nicely with the other change since I no longer need to assign `modifiedTime` to a variable